### DISCUSSION TITLE
Support changing the query on FirebaseAnimatedList

### DISCUSSION THREAD

**Discussion by amervelic** [firebase/flutterfire#6056] [Feature request]
When the `query` passed to `FirebaseAnimatedList` changes, the list keeps showing the results of the previous query instead.

Adding a `didUpdateWidget` to detect that the query changed should fix this issue.

**Comment by amenasyed09** [comment]
The problem is that FirebaseAnimatedList sometimes doesn't detect query changes properly and needs to be explicitly told to refresh its data.So try adding a Key ( for eg : Key _listKey = UniqueKey();)  .When updating your query, generate a new key to force a rebuild , and lastly add the key to your FirebaseAnimated list . Without this, the widget sometimes doesn't detect query changes and continues showing old results.
